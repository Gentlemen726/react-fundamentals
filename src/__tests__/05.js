import React from 'react'
import ReactDOMServer from 'react-dom/server.browser'
import prettyFormat from 'pretty-format'
import Usage from '../exercises-final/05'
// import Usage from '../exercises/05'

const {DOMElement, DOMCollection} = prettyFormat.plugins

// TODO: Improve this test...
expect.addSnapshotSerializer({
  test: val => typeof val === 'string',
  print: val => val,
})

test('renders the correct styles', () => {
  // we're doing this renderToStaticMarkup business because
  // for some reason rendering it like normally doesn't give
  // us the style props correctly.
  const html = ReactDOMServer.renderToStaticMarkup(<Usage />)
  const div = document.createElement('div')
  div.innerHTML = html

  const formattedHTML = prettyFormat(div.firstChild, {
    plugins: [DOMElement, DOMCollection],
  })
  expect(formattedHTML).toMatchSnapshot()
})
