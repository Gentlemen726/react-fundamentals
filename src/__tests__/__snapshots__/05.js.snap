// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the correct styles 1`] = `
<div>
  <div
    class="box box--small"
    style="background-color:lightblue"
  >
    small lightblue box
  </div>
  <div
    class="box box--medium"
    style="background-color:pink"
  >
    medium pink box
  </div>
  <div
    class="box box--large"
    style="background-color:orange"
  >
    large orange box
  </div>
</div>
`;
